src/filters.cpp

```diff
@@ -1,32 +1,22 @@
 #include "filters.hpp"
-
 #include <exception>
 #include <map>
-// This header must come before `mgbl` headers
-// So we ask clang-format not to re-order
-// clang-format off
-#include <node/src/node_conversion.hpp>
-// clang-format on
 #include <mbgl/style/conversion.hpp>
 #include <mbgl/style/conversion/filter.hpp>
+#include <mbgl/style/conversion/json.hpp>
 #include <mbgl/style/filter.hpp>
 #include <string>
 #include <utility>
 
-// Single constructor used for any new instance of it.
-auto Filters::constructor() -> Nan::Persistent<v8::FunctionTemplate>& {
-    static Nan::Persistent<v8::FunctionTemplate> init;
-    return init;
-} // LCOV_EXCL_LINE
-
-void Filters::Initialize(v8::Local<v8::Object> target) {
-    Nan::HandleScope scope;
-    v8::Local<v8::FunctionTemplate> lcons = Nan::New<v8::FunctionTemplate>(Filters::New);
-    lcons->InstanceTemplate()->SetInternalFieldCount(1);
-    lcons->SetClassName(Nan::New("Filters").ToLocalChecked());
-    Nan::SetPrototypeMethod(lcons, "layers", layers);
-    target->Set(Nan::New("Filters").ToLocalChecked(), lcons->GetFunction());
-    constructor().Reset(lcons);
+Napi::FunctionReference Filters::constructor; // NOLINT
+
+Napi::Object Filters::Initialize(Napi::Env env, Napi::Object exports) {
+    Napi::Function func = DefineClass(env, "Filters",
+                                      {InstanceMethod("layers", &Filters::layers)});
```

Comment:
How about we use the template version available in node-addon-api?

```
InstanceMethod<&Filters::layers>("layers")
```

Reply:
9e5807bbf78606843f4bf0fe7d356caabca95b7d